packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -330,18 +329,33 @@ const Panel: React.FC<SelectPanelProps> = ({
 
 const SelectPanelButton = React.forwardRef<HTMLButtonElement, ButtonProps>((props, anchorRef) => {
   const inputProps = useFormControlForwardedProps(props)
-  return (
-    <>
-      {inputProps.id && (
-        <VisuallyHidden id={`${inputProps.id}--select-panel-button-children`}>{inputProps.children}</VisuallyHidden>
-      )}
+  const [labelId, setLabelId] = useState('')
+  useEffect(() => {
+    const label = document.querySelector(`[for='${inputProps.id}']`)
+    if (label) {
+      if (label.id) {
+        setLabelId(label.id)
+      } else {
+        const newLabelId = `${inputProps.id}--select-panel-button-label`
+        label.id = newLabelId
+        setLabelId(newLabelId)
+      }
+    }
+  }, [inputProps.id])
+
+  const selectPanelButtonId = `${inputProps.id}--select-panel-button`
+  if (inputProps.id && labelId) {
+    return (
       <Button
         ref={anchorRef}
+        aria-labelledby={`${labelId} ${selectPanelButtonId}`}
```

Comment:
@TylerJDev, this looks correct. However, it this is read as one long strong: "SelectPanel within FormControl Choose a tag".

<img width="646" alt="screen reader preview with the text: SelectPanel within FormControl Choose a tag" src="https://github.com/primer/react/assets/1863771/96a5654d-4dcb-4b95-bebf-a3be737bcba6">

Should we artificially try to add a comma in there? "SelectPanel within FormControl, Choose a tag"

Reply:
Great point! I do think we should add something here, I feel that the comma would be a good option. It should provide a slight pause for most screen readers, such as with other punctuation like (`:`, `;`, etc).

Reply:
Is there a good way to do this? If I do
```tsx
aria-labelledby={`${labelId}, ${selectPanelButtonId}`}
```
It tries to find an element with the `id` `${labelId},` and can't so then it only ends up using the `${selectPanelButtonId}` label. The only way I can do it is by adding `VisuallyHidden` text with a comma and adding that as a third label in between the other two like:
```tsx
aria-labelledby={`${labelId}, ${selectPanelButtonId}`}
```
const selectPanelButtonId = `${inputProps.id}--select-panel-button`
  const punctuationId = `${inputProps.id}--select-panel-button-punctuation`
  const punctuation = ','
  if (inputProps.id && labelId) {
    return (
      <>
        <VisuallyHidden id={punctuationId}>{punctuation}</VisuallyHidden>
        <Button
          ref={anchorRef}
          aria-labelledby={`${labelId} ${punctuationId} ${selectPanelButtonId}`}
          {...inputProps}
          id={selectPanelButtonId}
        />
      </>
    )
```

Reply:
I think that's unfortunately the only way to do it 😢

Reply:
Alright I pushed that change. It also doesn't do "a, b" it does "a , b" so the spacing of the comma is a little bit off.

Reply:
Is it possible to add the comma to the end of the first label? So if `A` is "SelectPanel within FormControl", we'd append the comma to the end of the label, rather than having 3 separate references for `aria-labelledby`? That'd way we'd have only `A` and `B` being referenced.

Reply:
That's a good idea, but then wouldn't show a comma in the UI for the label as well?

Reply:
We could have it visually hidden as well, for example: 

```
<Label>
  SelectPanel within FormControl<VisuallyHidden>,</VisuallyHidden>
</Label>
```

This should still announce for screen readers, and will not be visually present within the label. In addition, if we could only add the comma conditionally when there's more than one value in the `aria-labelledby`, that would be best, as there's no need for punctuation if there's only one label.

Reply:
I'm able to do this
```tsx
      label.innerHTML = label.innerHTML.trim()
      label.innerHTML += `
        <span style="
          position: absolute;
          width: 1px;
          height: 1px;
          padding: 0;
          margin: -1px;
          overflow: hidden;
          clip: rect(0, 0, 0, 0);
          white-space: nowrap;
          border-width: 0;"
        >,</span>
      `
```
Although for some reason it still puts a space in between the label text and the ","
<img width="345" alt="image" src="https://github.com/primer/react/assets/29404461/fd977390-9d65-415b-b29e-fdcbc8fa3f99">
<img width="241" alt="image" src="https://github.com/primer/react/assets/29404461/98e5b9c8-eb9f-4c33-a52e-251fcf37b225">

There may be some method of removing this space, but either way it seems at the worst dangerous and the best very non-React to force set the innerHTML like that 🤔

Reply:
I don't think we need to remove the space - as long as the comma is at the end, and it's not affecting the visual look of the label, then the announcement should be fine with or without the space.

Reply:
Which approach do you think is better? 1) Manually setting the `innerHTML` or 2) having three values in `aria-labelledby` and one of them is `VisuallyHidden`? Personally I prefer the 2nd approach because the first approach seems like it could cause problems in edge cases when manually setting `innerHTML`.

Reply:
Let me run this by the Accessibility team, I'm not 100% sure if there's any implications we'd have from referencing three values, but if not, it'd be preferable to use this rather than setting `innerHTML` directly.

Reply:
Thanks! If you could cc me when you ask them that would be great, I'm interested in what are a11y practices are in this scenario 😄

Reply:
Are there any a11y issues with visually hiding punctuation like this and only using it as part of the aria-labelledby attribute?